ir.ts

```diff
@@ -18,22 +18,25 @@ export type Stmt<A> =
   | {  a?: A, tag: "pass" }
   | {  a?: A, tag: "ifjmp", cond: Value<A>, thn: string, els: string }
   | {  a?: A, tag: "jmp", lbl: string }
-
   | { a?: A, tag: "store", start: Value<A>, offset: Value<A>, value: Value<A> } // start should be an id
-
+  | { a?: A, tag: "store_str", start: Value<A>, offset: Value<A>, value: Expr<A> } // start should be an id
```

Comment:
Why can't the value be an identifier? The idea of lower is that you'd introduce any intermediate variables you need so that you can use `Value` in these positions.

Reply:
The reason is that the "store_str" is used in this situation:
```
x:str = "abcd"
c:str = ""
x = "xyz"
c = x[1]
```
In this case, in lower.ts we cannot get the value of ```x[1]``` directly. Therefore, we need to load it from the memory. So, we use the ```store_str``` expression which can receive an ```load``` expression as the value. In this way, we can use the same ```store``` function defined in memory.wat. 

I agree that the ```value``` of ```store_str``` can be a ```Value```. In the way, we still need the ```store_str``` and we need to write a new```store_str``` wat code which similar to the defined ```store``` but it will encapsulate the ```load``` expression in the wat level. 

I think both of the solution can achieve the goals. The reason why we choose the current solution is that it can reuse the ```store``` wat code without writing a new one.

Reply:
Oh I think I have got your point. We can create a new variable and use the ```assign``` stmt (which can receive ```Expr``` as its value) to assign the ascii code to it. Then we can use the original ```store``` in both ```ir.ts``` and ```memory.wat``` without further changes.

Reply:
fixed.